src/coreclr/vm/jithelpers.cpp

```diff
@@ -1386,11 +1386,18 @@ HCIMPL0(void, IL_Rethrow)
         RealCOMPlusThrow(kInvalidProgramException, (UINT)IDS_EE_RETHROW_NOT_ALLOWED);
     }
 
-    HELPER_METHOD_FRAME_END();
+    UNINSTALL_UNWIND_AND_CONTINUE_HANDLER;
+    UNINSTALL_MANAGED_EXCEPTION_DISPATCHER;
+#else // FEATURE_EH_FUNCLETS
+    PORTABILITY_ASSERT("IL_Rethrow");
+#endif // FEATURE_EH_FUNCLETS
+
+    FC_CAN_TRIGGER_GC_END();
```

Comment:
Just to make sure, it's not swapping one HMF wrapper to another HMF wrapper, right? Per https://github.com/dotnet/runtime/pull/109087#issuecomment-2435403360, `FC_GC_POLL_RET` expands to HMF.

Reply:
I'd hope so. I don't see `FC_GC_POLL_RET` anywhere anymore... and the `FC_CAN_TRIGGER_GC_END` part was already present in the `FEATURE_EH_FUNCLETS` branch.